hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java

```diff
@@ -3402,4 +3406,33 @@ public ListReplicationSinkServersResponse listReplicationSinkServers(
     }
     return builder.build();
   }
+
+  @Override
+  public RegionServerReportResponse replicationServerReport(RpcController controller,
+      RegionServerReportRequest request) throws ServiceException {
+    try {
+      master.checkServiceStarted();
+      int versionNumber = 0;
+      String version = "0.0.0";
+      VersionInfo versionInfo = VersionInfoUtil.getCurrentClientVersionInfo();
+      if (versionInfo != null) {
+        version = versionInfo.getVersion();
+        versionNumber = VersionInfoUtil.getVersionNumber(versionInfo);
+      }
+      ClusterStatusProtos.ServerLoad sl = request.getLoad();
+      ServerName serverName = ProtobufUtil.toServerName(request.getServer());
+      ServerMetrics oldMetrics = master.getReplicationServerManager().getServerMetrics(serverName);
+      ServerMetrics newMetrics =
+          ServerMetricsBuilder.toServerMetrics(serverName, versionNumber, version, sl);
+      master.getReplicationServerManager().serverReport(serverName, newMetrics);
+      if (sl != null && master.metricsMaster != null) {
+        // Up our metrics.
+        master.metricsMaster.incrementRequests(sl.getTotalNumberOfRequests()
```

Comment:
One new question: ReplicationServer should not have request count metric? Because this metric should mean that read metric or write metric?

Reply:
RegionServer increments request count metric in all RPC method, including `replicateWALEntry`, so the request count metric should mean more than just reading and writing metric?

Reply:
Got it.